### PR TITLE
Dutch National Flag Algorithm in java

### PR DESCRIPTION
**Pull request by Rajashri14**

### CODE REVIEW

**Comment by raksha009** [comment]
Plz add issue no

File: Code/Java/Segregate01and2.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+//Driver function to check for above functions.
+public class Segregate01and2
+{
+    public static void main(String[] args)
+    {
+        System.out.println("Dutch National Flag Algorithm. ");
+
+        int a[] = {1, 0, 2, 1, 0, 2, 1, 0, 0, 1,0};
```
Comment (sayantani11) [inline Code/Java/Segregate01and2.java:77]: Try to take user input

File: Code/Java/Segregate01and2.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package openSource;
+
+// Issue No - 958
+
+/*
+ Given an array consisting of 0s,1s and 2s.The task is to write a function that sorts the given array without using any sorting algorithm.
+ The function should put all 0s first,then all 1s and all 2s in last.
+*/
+
```
Comment (sayantani11) [inline Code/Java/Segregate01and2.java:9]: Add sample input here

File: Code/Java/Segregate01and2.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    public static void main(String[] args)
+    {
+        System.out.println("Dutch National Flag Algorithm. ");
+
+        int a[] = {1, 0, 2, 1, 0, 2, 1, 0, 0, 1,0};
+
+        Seg01and2 arr = new Seg01and2();
+        arr.Segregate01and2(a);
+        arr.print(a);
+
```
Comment (sayantani11) [inline Code/Java/Segregate01and2.java:82]: Remove extra spaces

File: Code/Java/Segregate01and2.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    }
+}
+
```
Comment (sayantani11) [inline Code/Java/Segregate01and2.java:86]: Add the sample output here

**Review by sayantani11 — CHANGES_REQUESTED** [review]

**Review by raksha009 — COMMENTED** [review]
Plz link your code to required file, Go through the guidelines and pinned issues.

File: Code/Java/Segregate01and2.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        arr.Segregate01and2(a);
+        arr.print(a);
+    }
+}
+// Sample Output - 0 0 0 0 0 1 1 1 1 2 2
+
+/*
+Time Complexity - O(n)
+Space Complexity - O(1)
+*/
```
Comment (raksha009) [inline Code/Java/Segregate01and2.java:91]: Add a new line here.

**Review by raksha009 — CHANGES_REQUESTED** [review]
Hey, plz go through the guidelines and the pinned issues and link the code with required folder.

**Comment by raksha009** [comment]
Are you still working on it ?

**Comment by raksha009** [comment]
> Plz add issue no

plz specify

File: Code/Java/Segregate01and2.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        for (int i=0 ; i< size;i++)
+        {
+            a[i]=sc.nextInt();
+        }
+        Seg01and2 arr = new Seg01and2();
+        arr.Segregate01and2(a);
+        arr.print(a);
+    }
+}
+// Sample Output - 0 0 0 0 0 1 1 1 1 2 2
```
Comment (raksha009) [inline Code/Java/Segregate01and2.java:83]: Follow 
Test Cases
Input : 
Ouput :

Add atleast 2 test cases 
Add add the test cases and complexities in a single multiline comment.

File: Code/Java/Segregate01and2.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
+
+/*
+ Given an array consisting of 0s,1s and 2s.The task is to write a function that sorts the given array.
+ The function should put all 0s first,then all 1s and all 2s in last but without using any sorting algorithm.
+*/
+
+// Sample Input - 1 0 2 1 0 2 1 0 0 1 0
```
Comment (raksha009) [inline Code/Java/Segregate01and2.java:7]: Remove this line

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by raksha009** [comment]
Issue #958

File: Code/Java/Segregate01and2.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+Test Cases :
+1)  Input  -  1 2 0 1 0 2
+    Output -  0 0 1 1 2 2
+   
+2)  Input  -   1 0 2 1 0 2 1 0 0 1 0
+    Output -   0 0 0 0 0 1 1 1 1 2 2
+
+Complexity Analysis :
+Time Complexity   -  O(n)
```
Comment (raksha009) [inline Code/Java/Segregate01and2.java:94]: Specify n

**Review by raksha009 — CHANGES_REQUESTED** [review]
Update special algo readme and not the java one readme. 
Just make these changes, will merge after this 👍🏼

**Pull request merged**